src/Entity/MajorVersion.php

```diff
@@ -282,33 +272,40 @@ public function toArray(): array
 
     public function isActive(): bool
     {
-        $dateTime = new \DateTimeImmutable();
-        return null === $this->getMaintainedUntil()
-            || $dateTime <= $this->getMaintainedUntil();
+        $dateTimeImmutable = new \DateTimeImmutable();
+        if (null === $this->getMaintainedUntil()) {
+            return true;
+        }
+        return $dateTimeImmutable <= $this->getMaintainedUntil();
```

Comment:
I disagree with both changes here, honestly.  What's Rector's reasoning?

Reply:
This is coming from the EarlyReturn rule. For me it increases the readability. Not sure if this also increases the processing time. Does PHP abort after the first positive evaluation in OR conditions? E.g. the Delphi compiler has an option to do so and is enabled by default.

Reply:
Yes, both OR and AND short-circuit.